newlib/libc/stdlib/random.c

```diff
@@ -54,15 +54,16 @@ algorithm as <<rand>>.
 
 #define _DEFAULT_SOURCE
 #include <stdlib.h>
+#include <stdint.h>
 
-__THREAD_LOCAL long long _rand_next = 1;
+__THREAD_LOCAL uint64_t _rand_next = 1;
 
 long int
 random (void)
 {
   /* This multiplier was obtained from Knuth, D.E., "The Art of
      Computer Programming," Vol 2, Seminumerical Algorithms, Third
      Edition, Addison-Wesley, 1998, p. 106 (line 26) & p. 108 */
-  _rand_next = _rand_next * __extension__ 6364136223846793005LL + 1;
+  _rand_next = _rand_next * __extension__ 6364136223846793005ULL + 1;
```

Comment:
Is it necessary to add `__extension__` since modern C compilers should handle `ULL` fine?

Reply:
yeah, long long was in C99. Feel free to remove that.